### PR TITLE
Issue #3

### PR DESCRIPTION
**Pull request by andreypanasenko**
Move example classes to specific package.
Made Serialization as interface and add Jackson serializer as parameter in ServerConfig constructor.
Add unit test for JacksonSerializer.
Add Integration test for operation with one node, upsert/get/delete/getAll operations.
Add Integration test for ZooKeeper cluster, upsert/delete on different nodes.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -138,7 +138,12 @@
             <version>${junit.version}</version>
             <scope>test</scope>
         </dependency>
-
+		<dependency>
+		   <groupId>org.apache.activemq</groupId>
+		   <artifactId>activemq-all</artifactId>
```
Comment (antonkharenko) [inline pom.xml:143]: why we need activemq?

File: src/main/java/com/ogp/configurator/ConfigService.java [hunk 2]
```diff
@@ -63,61 +66,73 @@ private ConfigService(Builder builder) {
 		try {
 			configCache.start();
 		} catch (Exception e) {
-			e.printStackTrace();
+			logger.error("Failed to TreeCache", e);
 			throw new RuntimeException();
```
Comment (antonkharenko) [inline src/main/java/com/ogp/configurator/ConfigService.java]: Lets use Throwables.propagate(e) from guava

File: src/main/java/com/ogp/configurator/ConfigService.java [hunk 3]
```diff
@@ -236,18 +260,18 @@ public Object getConfigEntity(Class<Object> clazz, String key) {
...
 		if (configObjects.containsKey(classFromType)) {
-			System.out.printf("Config cache for class %s have %d elements\n", classFromType.toString(), configObjects.get(classFromType).size());
+			logger.debug("Config cache for class %s have %d elements\n", classFromType.toString(), configObjects.get(classFromType).size());
 			Map<String, Object> entities = configObjects.get(classFromType);
 			synchronized (entities) {
 				configEntities.addAll(entities.values());
 			}
 		} else {
-			System.out.printf("Class %s not found in cache\n", classFromType);
+			logger.debug("Class %s not found in cache\n", classFromType);
```
Comment (antonkharenko) [inline src/main/java/com/ogp/configurator/ConfigService.java]: Does it work like this? [maybe should be - logger.debug("Class {} not found in cache", classFromType); ]

File: src/main/java/com/ogp/configurator/ConfigService.java [hunk 4]
```diff
@@ -63,61 +66,73 @@ private ConfigService(Builder builder) {
...
+					}
+				}
+				break;
+			case INITIALIZED:
+				reloadConfigTree();
+				logger.info("Initialization complete");
+				isInitialized = true;
+				isReadOnly = false;
+				synchronized (syncReady) {
+					syncReady.notify();
```
Comment (antonkharenko) [inline src/main/java/com/ogp/configurator/ConfigService.java]: Lets not to use Java low level synchronization primitives, but more high level utils from java concurrency (e.g. count down latch)

File: src/main/java/com/ogp/configurator/ConfigService.java [hunk 5]
```diff
@@ -166,12 +181,36 @@ private void initPaths() {
 		}
 	}
 
+	public void BlockUntilReady() {
```
Comment (antonkharenko) [inline src/main/java/com/ogp/configurator/ConfigService.java]: start from lower case... also lets rename to "awaitInitialization"

File: src/main/java/com/ogp/configurator/serializer/ISerializer.java [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package com.ogp.configurator.serializer;
+
+public interface ISerializer {
+
+	public byte[] Serialize(Object obj) throws Exception;
```
Comment (antonkharenko) [inline src/main/java/com/ogp/configurator/serializer/ISerializer.java]: method name start from lower case

File: pom.xml (hunk 1 above)
Comment (andreypanasenko) [inline pom.xml:143]: Curator Cluster Test use it.

File: src/main/java/com/ogp/configurator/ConfigService.java (hunk 5 above)
Comment (andreypanasenko) [inline src/main/java/com/ogp/configurator/ConfigService.java]: rename to "awaitConnected"

File: src/main/java/com/ogp/configurator/ConfigService.java (hunk 3 above)
Comment (andreypanasenko) [inline src/main/java/com/ogp/configurator/ConfigService.java]: Fixed

File: src/main/java/com/ogp/configurator/serializer/ISerializer.java (hunk 6 above)
Comment (andreypanasenko) [inline src/main/java/com/ogp/configurator/serializer/ISerializer.java]: Changed

**Comment by antonkharenko** [comment]
+1

**Pull request merged**